src/Controller/ResetPasswordControllerTrait.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SymfonyCasts\Bundle\ResetPassword\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+/**
+ * Provides useful methods to a "reset password controller"
+ *
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+trait ResetPasswordControllerTrait
+{
+    private function setCanCheckEmailInSession(Request $request, bool $value = true): void
```

Comment:
I don't think `$value` needs to be configurable.

AND! We can avoid the user needing to pass in the request :). We will assume (and if people don't like it, "tough" for them - they can avoid the trait ;) ) that the controller extends `AbstractController` (probably we should mention that in the class summary). In that case, the request_stack is an injected service, meaning we can say:

```php
$request = $this->container->get('request_stack')->getCurrentRequest()
```

Reply:
done